packages/vite/client.d.ts

```diff
@@ -47,6 +67,10 @@ interface ImportMetaEnv {
   PROD: boolean
 }
 
+// See https://stackoverflow.com/a/63549561.
+type CustomEventName<T extends string> = (T extends `vite:${T}` ? never : T) &
+  (`vite:${T}` extends T ? never : T)
```

Comment:
Mh 🤔 if custom events from plugins can also be send, should we enforce the type to start with `vite:` always?

@patak-js What do you think?

Otherwise I now approve this PR from my side

Reply:
I don't know if I understand the question, we should enforce `vite:` for our events, and custom events are free to use their own naming convention.

Reply:
I don't know if I understand the question, we should enforce `vite:` for our events, and custom events are free to use their own naming convention.